internal/metastructure/metastructure.go

```diff
@@ -495,6 +495,10 @@ func (m *Metastructure) DestroyForma(forma *pkgmodel.Forma, config *config.Forma
 
 func (m *Metastructure) DestroyByQuery(query string, config *config.FormaCommandConfig, clientID string) (*apimodel.SubmitCommandResponse, error) {
 	q := querier.NewBlugeQuerier(m.Datastore)
+	if err := q.ValidateDestroy(query); err != nil {
```

Comment:
Should we introduce a `DestroyResourcesQuery`, like we have the `StatusQuery`? The query could then validate itself during the field assignment, similar to [how we do this in the status query]( https://github.com/platform-engineering-labs/formae/blob/main/internal/metastructure/querier/bluge_querier.go#L94-L119). That way we don't have two ways of validating queries, and it'll be a lot less code as well.

When we're fixing this we should also ensure that all errors are correctly wrapped in an `InvalidQueryError` so they get rendered correctly on the [CLI](https://github.com/platform-engineering-labs/formae/blob/main/internal/cli/renderer/errors.go#L63-L65). When I was looking at the issue I noticed that in a bunch of places we we are not doing this.